numba_dpex/tests/core/types/range_types/test_unbox_box.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-FileCopyrightText: 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from numba_dpex import NdRange, Range, dpjit
+
+ranges = [(10,), (10, 10), (10, 10, 10)]
+
+
+@pytest.mark.parametrize("r", ranges)
+def test_range_unbox_box(r):
+    @dpjit
+    def _tester(r):
+        return r
+
+    r_in = Range(*r)
+    r_out = _tester(r_in)
+
+    assert r_out.ndim == r_in.ndim
```

Comment:
May be we should compare objects and overwrite `_eq` if needed?
```suggestion
    assert r_in == r_out
```

Reply:
Done in [084cd8b](https://github.com/IntelPython/numba-dpex/pull/1148/commits/084cd8b4785030c2a177b5d292ed8b05a1dc887a)